tests/fe/fe_test.h

```diff
@@ -0,0 +1,232 @@
+#ifndef __fe_test_h__
+#define __fe_test_h__
+
+#include "test_comm.h"
+
+#include <libmesh/dof_map.h>
+#include <libmesh/elem.h>
+#include <libmesh/equation_systems.h>
+#include <libmesh/fe_base.h>
+#include <libmesh/fe_interface.h>
+#include <libmesh/mesh.h>
+#include <libmesh/mesh_generation.h>
+#include <libmesh/numeric_vector.h>
+#include <libmesh/system.h>
+
+#include <cppunit/extensions/HelperMacros.h>
+#include <cppunit/TestCase.h>
+
+#include <vector>
+
+#define FETEST                   \
+  CPPUNIT_TEST( testU );         \
+  CPPUNIT_TEST( testGradU );     \
+  CPPUNIT_TEST( testGradUComp );
+
+using namespace libMesh;
+
+Number linear_test (const Point& p,
+                    const Parameters&,
+                    const std::string&,
+                    const std::string&)
+{
+  const Real & x = p(0);
+  const Real & y = (LIBMESH_DIM > 1) ? p(1) : 0;
+  const Real & z = (LIBMESH_DIM > 2) ? p(2) : 0;
+
+  return x + 0.25*y + 0.0625*z;
+}
+
+
+template <Order order, FEFamily family, ElemType elem_type>
+class FETest : public CppUnit::TestCase {
+
+private:
+  unsigned int _dim, _nx, _ny, _nz;
+  Elem *_elem;
+  std::vector<dof_id_type> _dof_indices;
+  FEBase * _fe;
+  Mesh* _mesh;
+  System* _sys;
+  EquationSystems* _es;
```

Comment:
Three different kinds of spacing for pointer declarations used in these 7 lines.